File: 문제풀이/온라인 저지/백준/# 04. Bronze Ⅱ/# 02864. 5와 6의 차이/java/Main.java
import java.io.DataInputStream;
import java.io.DataOutputStream;

public class Main {

    public static void main(String[] args) {
        int[] a = i();
        int[] b = i();
        writeInt(a[0] + b[0]);
        writeInt(a[1] + b[1]);
        flushBuffer();
    }

    private static final byte EOF = -1;
    private static final byte ASCII_space = 32;
    private static final byte ASCII_minus = 45;
    private static final byte ASCII_0 = 48;
    private static final DataInputStream din = new DataInputStream(System.in);
    private static final DataOutputStream dout = new DataOutputStream(System.out);
    private static final byte[] inbuffer = new byte[16];
    private static int inbufferpointer;
    private static int bytesread;
    private static final byte[] outbuffer = new byte[16];
    private static int outbufferpointer;
    private static final byte[] bytebuffer = new byte[16];

    private static int[] i() {
        int min = 0;
        int max = 0;
        byte c;
        while (isDigit(c = r())) {
            min = min * 10 + ((c == 54) ? 5 : c - 48);
            max = max * 10 + ((c == 53) ? 6 : c - 48);
        }
        return new int[]{min, max};
    }

    private static byte r() {
        if (inbufferpointer == bytesread) fillBuffer();
        return (bytesread == EOF) ? EOF : inbuffer[inbufferpointer++];
    }

    private static void fillBuffer() {
        try {
            bytesread = din.read(inbuffer, (inbufferpointer = 0), inbuffer.length);
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }

    private static void writeInt(int i) {
        if (i == 0) {
            writeByte(ASCII_0);
        } else {
            if (i < 0) {
                writeByte(ASCII_minus);
                i = -i;
            }
            int index = 0;
            while (i > 0) {
                bytebuffer[index++] = (byte) (i % 10 + ASCII_0);
                i /= 10;
            }
            while (index-- > 0) {
                writeByte(bytebuffer[index]);
            }
        }
        writeByte(ASCII_space);
    }

    private static void writeByte(byte b) {
        if (outbufferpointer == outbuffer.length) flushBuffer();
        outbuffer[outbufferpointer++] = b;
    }

    private static void flushBuffer() {
        if (outbufferpointer != 0) {
            try {
                dout.write(outbuffer, 0, outbufferpointer);
            } catch (Exception e) {
                throw new RuntimeException(e);
            }
            outbufferpointer = 0;
        }
    }

    private static boolean isDigit(byte b) {
        return b >= ASCII_0;
    }
}

File: 문제풀이/온라인 저지/백준/# 04. Bronze Ⅱ/# 02864. 5와 6의 차이/kotlin/Main.kt
fun main() {
    val a = i()
    val b = i()
    writeInt(a[0] + b[0])
    writeInt(a[1] + b[1])
    flushBuffer()
}

private const val EOF = -1
private const val ASCII_space: Byte = 32
private const val ASCII_minus: Byte = 45
private const val ASCII_0: Byte = 48
private val din = java.io.DataInputStream(System.`in`)
private val dout = java.io.DataOutputStream(System.out)
private val inbuffer = ByteArray(16)
private var inbufferpointer = 0
private var bytesread = 0
private val outbuffer = ByteArray(16)
private var outbufferpointer = 0
private val bytebuffer = ByteArray(16)

private fun i(): IntArray {
    var min = 0
    var max = 0
    var c: Int
    while (isDigit(r().also { c = it.toInt() })) {
        min = min * 10 + if (c == 54) 5 else c - 48
        max = max * 10 + if (c == 53) 6 else c - 48
    }
    return intArrayOf(min, max)
}

private fun r(): Byte {
    if (inbufferpointer == bytesread) fillBuffer()
    return if (bytesread == EOF) EOF.toByte() else inbuffer[inbufferpointer++]
}

private fun fillBuffer() {
    bytesread = din.read(inbuffer, 0.also { inbufferpointer = it }, inbuffer.size)
}

private fun writeInt(i: Int) {
    var i = i
    if (i == 0) {
        writeByte(ASCII_0)
    } else {
        if (i < 0) {
            writeByte(ASCII_minus)
            i = -i
        }
        var index = 0
        while (i > 0) {
            bytebuffer[index++] = (i % 10 + ASCII_0).toByte()
            i /= 10
        }
        while (index-- > 0) {
            writeByte(bytebuffer[index])
        }
        writeByte(ASCII_space)
    }
}

private fun writeByte(b: Byte) {
    if (outbufferpointer == outbuffer.size) flushBuffer()
    outbuffer[outbufferpointer++] = b
}

private fun flushBuffer() {
    if (outbufferpointer != 0) {
        dout.write(outbuffer, 0, outbufferpointer)
        outbufferpointer = 0
    }
}

private fun isDigit(b: Byte) = b >= ASCII_0